hazelcast/src/main/java/com/hazelcast/internal/memory/MemoryAccessStrategy.java

```diff
@@ -0,0 +1,576 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.memory;
+
+import java.lang.reflect.Field;
+
+/**
+ * Abstract entity which provides different way to access data from external resources
+ *
+ * @param <R> - type of resource
+ */
+public interface MemoryAccessStrategy<R> extends ByteMemoryAccessStrategy<R> {
+    MemoryAccessStrategy<Object> STANDARD_MEM = MemoryAccessorStrategyProvider.getMemoryAccessStrategy(MemoryAccessStrategyType.STANDARD);
+
+    MemoryAccessStrategy<Object> STANDARD_ALIGNED_MEM = MemoryAccessorStrategyProvider.getMemoryAccessStrategy(MemoryAccessStrategyType.STANDARD_ALIGNMENT);
+
+    MemoryAccessStrategy<byte[]> HEAP_BYTE_ARRAY_MEM = MemoryAccessorStrategyProvider.getMemoryAccessStrategy(MemoryAccessStrategyType.HEAP_BYTE_ARRAY);
+
+    MemoryAccessStrategy<Object> MEM = MemoryAccessorStrategyProvider.getMemoryAccessStrategy(MemoryAccessStrategyType.PLATFORM_AWARE);
```

Comment:
Retracting the above comment... I thought `MemoryAccessStrategy` would replace the current usages of `MEM` and `AMEM`, but these will remain `MemoryAccesor`s. I guess then the name `MEM` should not be reused.